File: document-ui/src/main/java/dev/sayaya/handbook/client/usecase/DocumentRepository.java
package dev.sayaya.handbook.client.usecase;

import dev.sayaya.handbook.client.domain.DocumentValue;
import dev.sayaya.rx.Observable;

import java.util.List;

/** 문서 저장/삭제/조회 포트 인터페이스. */
public interface DocumentRepository {
    Observable<DocumentValue[]> search(String type, int page, int limit);
    Observable<Void> save(List<DocumentValue> documents);
    Observable<Void> delete(List<DocumentValue> documents);
}

File: search-document/src/main/kotlin/dev/sayaya/handbook/usecase/DocumentRepository.kt
package dev.sayaya.handbook.usecase

import dev.sayaya.handbook.domain.Document
import dev.sayaya.handbook.domain.Search
import org.springframework.data.domain.Page
import reactor.core.publisher.Mono
import java.time.Instant
import java.util.*

/** 문서 읽기 전용 포트. */
interface DocumentRepository {
    fun search(workspace: UUID, param: Search): Mono<Page<Document>>
    fun find(workspace: UUID, type: String, serial: String, date: Instant): Mono<Document>
}
